### ISSUE TITLE
Bad Process.StartTime in .NET Core 3.1 on macOS

### ISSUE THREAD

**Issue by mattjohnsonpint** [dotnet/runtime#66170]
### Description

The value returned from `System.Diagnostics.Process.StartTime` is wildly incorrect when run on .NET Core 3.1 (runtime 3.1.22) on macOS 12.2.1 with M1 (Apple Silicon / arm64) processor.  The problem does *not* occur on .NET 6.0 or 5.0, nor does it occur on .NET Core 3.1 running on macOS with an Intel (x64) processor.

### Reproduction Steps

Following [this guidance](https://github.com/dotnet/sdk/issues/22380#issuecomment-964044326), install the following SDKs on macOS:
- 6.0.200 arm64
- 6.0.200 x64
- 5.0.405 x64
- 3.1.416 x64

Create a new console app:

test.csproj:
```csproj
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFrameworks>net6.0;net5.0;netcoreapp3.1</TargetFrameworks>
  </PropertyGroup>
</Project>
```

Program.cs:
```csharp
using System;
using System.Diagnostics;
using System.Runtime.InteropServices;

namespace ConsoleApp1
{
    public static class Program
    {
        public static void Main()
        {
            var processStartTime = Process.GetCurrentProcess().StartTime.ToUniversalTime();
            var utcNow = DateTime.UtcNow;
            var delta = utcNow - processStartTime;

            Console.WriteLine($"OS Architecture: {RuntimeInformation.OSArchitecture}");
            Console.WriteLine($"Framework: {RuntimeInformation.FrameworkDescription}");
            Console.WriteLine($"Process.StartTime: {processStartTime:o}");
            Console.WriteLine($"DateTime.UtcNow: {utcNow:o}");
            Console.WriteLine($"Delta: {delta}");
            Console.WriteLine();
        }
    }
}
```

Run the program under the various runtimes and observe the results.

### Expected behavior

The process start time should be very close to `DateTime.UtcNow` (delta near zero) in all cases.

### Actual behavior

The output is correct on .NET 6 and 5, but wildly incorrect on .NET Core 3.1 (LTS).

```
% dotnet run -f net6.0
OS Architecture: Arm64
Framework: .NET 6.0.2
Process.StartTime: 2022-03-03T22:26:01.6263462Z
DateTime.UtcNow: 2022-03-03T22:26:01.7393750Z
Delta: 00:00:00.1130288

% dotnet run -f net6.0 -a x64
OS Architecture: X64
Framework: .NET 6.0.2
Process.StartTime: 2022-03-03T22:26:07.7978340Z
DateTime.UtcNow: 2022-03-03T22:26:07.9519200Z
Delta: 00:00:00.1540860

% dotnet run -f net5.0       
OS Architecture: X64
Framework: .NET 5.0.14
Process.StartTime: 2022-03-03T22:26:18.0786000Z
DateTime.UtcNow: 2022-03-03T22:26:18.2399030Z
Delta: 00:00:00.1613030

% dotnet run -f netcoreapp3.1
OS Architecture: X64
Framework: .NET Core 3.1.22
Process.StartTime: 2022-03-03T06:58:55.5796520Z
DateTime.UtcNow: 2022-03-03T22:26:51.5452700Z
Delta: 15:27:55.9656180
```

### Regression?

The symptoms are similar to #19928, but that was resolved very long ago in .NET Core 2.0.

That it is working in .NET 5 and 6, might be related to #30241

**Comment by bruno-garcia** [comment]
I'm running macOS x64 and I get:

```
➜ dotnet run -f netcoreapp3.1
OS Architecture: X64
Framework: .NET Core 3.1.22
Process.StartTime: 2022-03-03T23:37:39.8810980Z
DateTime.UtcNow: 2022-03-03T23:37:40.9433880Z
Delta: 00:00:01.0622900

➜ dotnet run -f net5.0       
OS Architecture: X64
Framework: .NET 5.0.13
Process.StartTime: 2022-03-03T23:37:51.3093380Z
DateTime.UtcNow: 2022-03-03T23:37:52.1889820Z
Delta: 00:00:00.8796440

➜ dotnet run -f net6.0
OS Architecture: X64
Framework: .NET 6.0.1
Process.StartTime: 2022-03-03T23:37:56.5150970Z
DateTime.UtcNow: 2022-03-03T23:37:56.5868800Z
Delta: 00:00:00.0717830
```

```
dotnet --info
.NET SDK (reflecting any global.json):
 Version:   6.0.101
 Commit:    ef49f6213a

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  11.0
 OS Platform: Darwin
 RID:         osx.10.16-x64
 Base Path:   /usr/local/share/dotnet/sdk/6.0.101/

Host (useful for support):
  Version: 6.0.1
  Commit:  3a25a7f1cc

.NET SDKs installed:
  2.1.818 [/usr/local/share/dotnet/sdk]
  3.1.414 [/usr/local/share/dotnet/sdk]
  3.1.415 [/usr/local/share/dotnet/sdk]
  3.1.416 [/usr/local/share/dotnet/sdk]
  5.0.402 [/usr/local/share/dotnet/sdk]
  5.0.403 [/usr/local/share/dotnet/sdk]
  5.0.404 [/usr/local/share/dotnet/sdk]
  6.0.100-rc.2.21505.57 [/usr/local/share/dotnet/sdk]
  6.0.100 [/usr/local/share/dotnet/sdk]
  6.0.101 [/usr/local/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.30 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.30 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.20 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.21 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.22 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.11 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.12 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.13 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.0-rc.2.21480.10 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.0 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.1 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.30 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.20 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.21 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.22 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.11 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.12 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.13 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.0-rc.2.21480.5 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.0 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.1 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

```

**Comment by mattjohnsonpint** [comment]
Thanks @bruno-garcia.  I updated the issue to reflect that it only occurs on M1 hardware, not on Intel.

**Comment by Jozkee** [comment]
@danmoseley @jeffhandley before investing time into porting/ensuring that https://github.com/dotnet/corefx/pull/39572 fixes this issue, can you say if this would meet the bar for 3.1 servicing?

**Comment by tarekgh** [comment]
@mattjohnsonpint it is nice to see you around github :-)

I am wondering if this is something really blocking you?  or can migrate to .NET 6.0 to work around this issue?

**Comment by danmoseley** [comment]
> occurs on M1 hardware, not on Intel.

That's curious, it's not obvious to me that dotnet/corefx#39572 is Arm64 specific.

**Comment by mattjohnsonpint** [comment]
Hi @tarekgh!  I'm working at Sentry now, on their client SDKs with @bruno-garcia.  This came up because `Process.StartTime` is used in a core path that our unit tests hit, and we support clients that are building for many versions of .NET.  We're working around the test failures for now by skipping, but that's just burying the issue.  In reality, we have customers who have built .NET Core 3.1 apps, and may be deploying them to newer hardware, such as the M1 MacBooks.  We'll have erroneous data for that subset of customers using Sentry unless this is fixed in the runtime.

**Comment by Jozkee** [comment]
Linking similar Linux issue https://github.com/dotnet/runtime/issues/64742.

**Comment by jeffhandley** [comment]
@Jozkee This would be a good servicing candidate because of how it can affect data in production scenarios that @mattjohnsonpint described. Before we bring it to tactics, we'll want to make sure we can reliably reproduce the issue and verify the fix. If feasible, getting validation from @mattjohnsonpint or @bruno-garcia against a local build from a branch would help address other hesitations too.

By the way, thanks for the excellent issue submission and minimal repro, @mattjohnsonpint!

**Comment by mattjohnsonpint** [comment]
Just wanted to check in on this.  Is it planned to be addressed for 3.1 servicing?  Thanks.

**Comment by adamsitnik** [comment]
I hate to say it, but we have been postponing the fix so long that 3.1 got out of support. Since all supported versions are not affected, I am going to close the issue.
